### PR TITLE
completable future Java 19 methods

### PR DESCRIPTION
**Pull request by njr-11**
Implement and test the new methods that were added to CompletableFuture (indirectly via Future) in Java 19.

### CODE REVIEW

**Comment by njr-11** [comment]
#build

File: dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/internal/ManagedCompletableFuture.java [hunk 1]
```diff
@@ -911,6 +909,20 @@ public CompletableFuture<T> exceptionallyComposeAsync(Function<Throwable, ? exte
         }
     }
 
+    // Java 19+
+    @FFDCIgnore({ CancellationException.class, CompletionException.class })
+    public Throwable exceptionNow() {
```
Comment (gjwatts) [inline dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/internal/ManagedCompletableFuture.java:914]: Isn't exceptionNow() supposed to call both isDone() and get()? -> https://download.java.net/java/early_access/jdk19/docs/api/java.base/java/util/concurrent/Future.html#exceptionNow()

Comment (njr-11) [reply dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/internal/ManagedCompletableFuture.java:914]: No, the JavaDoc doesn't say that and only mentions the default implementation taking that approach.  A different approach is more optimal for completable futures.

Comment (gjwatts) [reply dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/internal/ManagedCompletableFuture.java:914]: Ok, thank you.

**Review by gjwatts — APPROVED** [review]
I am giving my approval for the first two files:
```
dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/internal/ManagedCompletableFuture.java
dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/internal/ManagedCompletionStage.java
```
Someone else will review:
```
dev/com.ibm.ws.concurrent_fat_jakarta/test-applications/ConcurrencyTestWeb/src/test/jakarta/concurrency/web/ConcurrencyTestServlet.java
```

File: dev/com.ibm.ws.concurrent_fat_jakarta/test-applications/ConcurrencyTestWeb/src/test/jakarta/concurrency/web/ConcurrencyTestServlet.java [hunk 2]
```diff
@@ -1092,6 +1096,149 @@ public void testEJBDDManagedThreadFactoryDefinition() throws Exception {
...
+                if (!(x.getCause() instanceof IllegalStateException))
+                    throw x;
+            } catch (IllegalStateException x) {
+                // pass
+            }
+
+            // Future.resultNow on cancelled task:
+            assertEquals(true, task3future.cancel(true));
+            try {
+                exception = (Throwable) exceptionNow.invoke(task3future);
```
Comment (KyleAure) [inline dev/com.ibm.ws.concurrent_fat_jakarta/test-applications/ConcurrencyTestWeb/src/test/jakarta/concurrency/web/ConcurrencyTestServlet.java:1211]: Comment say's `resultNow` but we are still testing `exceptionNow`. 
I assume this is a typo in the comment, but if you do intend to test resultNow the test will need to be updated.

**Review by KyleAure — APPROVED** [review]

File: dev/com.ibm.ws.concurrent_fat_jakarta/test-applications/ConcurrencyTestWeb/src/test/jakarta/concurrency/web/ConcurrencyTestServlet.java (hunk 2 above)
Comment (njr-11) [reply dev/com.ibm.ws.concurrent_fat_jakarta/test-applications/ConcurrencyTestWeb/src/test/jakarta/concurrency/web/ConcurrencyTestServlet.java:1211]: Yes, the comment is a mistake (actually a copy/paste error). Thanks for spotting it.

**Pull request merged**